lib/dom.js

```diff
@@ -831,7 +863,6 @@ Document.prototype = {
 		node.nodeName = tagName;
 		node.tagName = tagName;
 		node.localName = tagName;
-		node.childNodes = new NodeList();
```

Comment:
That makes a lot of sense. But is there a specific reason to move it as part of this PR?
If not, would you mind to file a separate PR for just this refactoring ...